vnext/JSI/Shared/ChakraCoreRuntime.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef CHAKRACORE
+
+#include "ChakraCoreRuntimeArgs.h"
+#include "ChakraRuntime.h"
+
+#include "ChakraCoreDebugger.h"
+
+#include <memory>
+#include <string>
+
+namespace Microsoft::JSI {
+
+class ChakraCoreRuntime : public ChakraRuntime {
+ public:
+  ChakraCoreRuntime(const std::shared_ptr<ChakraCoreRuntimeArgs> &args) noexcept;
```

Comment:
>std::shared_ptr<ChakraCoreRuntimeArgs> & [](start = 26, length = 40)

(nit) Since this is a constructor, would it make sense to pass by std::shared_ptr<ChakraCoreRuntimeArgs>&&?